### PR TITLE
Fix callable-types should handle `new(): T` just like `(): T`

### PR DESCRIPTION
**Pull request by rrogowski**
#### PR checklist

- [x] Addresses an existing issue: #4291 
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [ ] Documentation update

#### Overview of change:

Walker for `callable-types` rule handles the case where an interface contains a single construct signature declaration.

#### CHANGELOG.md entry:

[bugfix][`callable-types`](https://palantir.github.io/tslint/rules/callable-types/) support interfaces containing a single construct signature (#4291)

### CODE REVIEW

**Comment by rrogowski** [comment]
@giladgray The CI lint tests pass with the local `bin/tslint`, but fail with the global `tslint`, since I have changed the expected behavior for the purposes of this bug fix. Any ideas how I can resolve the failing tests? I'm a newer contributor to this project, so I'm not sure what steps I need to take.

**Comment by rrogowski** [comment]
@giladgray Lint passing. Please see my newly added TODO item in `src/language/formatter/formatter.ts`.

File: src/language/formatter/formatter.ts [hunk 1]
```diff
@@ -46,7 +46,11 @@ export interface IFormatterMetadata {
 
 export type ConsumerType = "human" | "machine";
 
+// TODO: Enable the `callable-types` rule below. Currently, it
+// produces a lint error in several locations complaining about the
+// `no-inferred-empty-object-type` rule.
```
[the remark this answers is not in the document]
Comment (rrogowski) [reply src/language/formatter/formatter.ts]: @JoshuaKGoldberg An issue has been filed (https://github.com/palantir/tslint/issues/4364) and I've added a line above the TODO linking to the issue.

**Comment by rrogowski** [comment]
@JoshuaKGoldberg An issue has been filed (https://github.com/palantir/tslint/issues/4364) and I've added a line above the TODO linking to the issue.

**Comment by ericanderson** [comment]
When I check out 02c4c9fa7a6897400daba1dc3d935f0ee1c05308, the tests dont seem to pass.

**Comment by ericanderson** [comment]
Prior tests seem to work. I'm going to revert this change.

**Pull request merged**